components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -0,0 +1,743 @@
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDO;
+import org.wso2.carbon.identity.oauth.endpoint.exception.CibaAuthFailureException;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.identity.openidconnect.model.Constants;
+
+import java.text.ParseException;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+
+/**
+ * Handles the validation of ciba authentication request.
+ */
+public class CibaAuthRequestValidator {
+
+    private static final Log log = LogFactory.getLog(CibaAuthRequestValidator.class);
+
+    /**
+     * Validate CIBA Authentication Request.
+     *
+     * @param request CIBA Authentication Request.
+     * @throws CibaAuthFailureException CIBA Authentication Failed Exception.
+     * @throws CibaAuthFailureException CIBA server serror.
+     */
+    public void validateAuthRequestParams(String request) throws CibaAuthFailureException {
+
+        try {
+            long timeInMillis = Calendar.getInstance(TimeZone.getTimeZone(CibaConstants.UTC)).getTimeInMillis();
+            long skewTime = OAuthServerConfiguration.getInstance().getTimeStampSkewInSeconds() *
+                    CibaConstants.SEC_TO_MILLISEC_FACTOR;
+            SignedJWT signedJWT = SignedJWT.parse(request);
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+
+            if (!isValidSignature(signedJWT)) {
+                // Signature is invalid.
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST, "Invalid signature.");
+            }
+
+            // Validate audience of the Request.
+            validateAudience(claimsSet);
+
+            // Validate  JWT-ID of the Request.
+            validateJti(claimsSet);
+
+            // Validate the expiryTime of the Request.
+            validateExpiryTime(claimsSet, timeInMillis, skewTime);
+
+            // Validate the issuedTime of the Request.
+            validateIssuedTime(claimsSet, timeInMillis);
+
+            // Validate the NBF of the Request.
+            validateNBF(claimsSet, timeInMillis, skewTime);
+
+            // Validate the scope of the Request.
+            validateScopes(claimsSet);
+
+            // Validate the client_notification_token of the Request.
+            validateACR(claimsSet);
+
+            // Validate the binding_message of the Request.
+            validateBindingMessage(claimsSet);
+
+            // Validate the transaction_context of the Request.
+            valiateTransactionContext(claimsSet);
+
+            // Validate the requested_expiry of the Request.
+            validateRequestedExpiry(claimsSet);
+
+            if (log.isDebugEnabled()) {
+                log.debug(" CIBA Authentication Request made by client with clientID : " +
```

Comment:
Remove the unnecessary space "CIBA Authentication Request

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1701